src/tests/java/picard/sam/markduplicates/QuerySortedMarkDuplicatesTest.java

```diff
@@ -42,4 +54,41 @@ protected boolean markUnmappedRecordsLikeTheirMates() {
 
     @Override
     protected AbstractMarkDuplicatesCommandLineProgramTester getTester() { return new QuerySortedMarkDuplicatesTester(); }
+
+    private class ByQualityQuerySortedTester extends AbstractMarkDuplicatesCommandLineProgramTester {
+
+        public ByQualityQuerySortedTester() {
+            super(DuplicateScoringStrategy.ScoringStrategy.SUM_OF_BASE_QUALITIES, SAMFileHeader.SortOrder.queryname);
+        }
+
+        @Override
+        protected CommandLineProgram getProgram() {
+            return new MarkDuplicates();
+        }
+    }
+
+    @DataProvider
+    public Object[][] SameUnclipped5primeEnd() {
+        final File TEST_DIR = new File("testdata/picard/sam/MarkDuplicates");
+        return new Object[][]{
+
+            new Object[]{new File(TEST_DIR, "SameUnclipped5primeEndv1.sam")},
+            new Object[]{new File(TEST_DIR, "SameUnclipped5primeEndv2.sam")}};
+    }
+
+    @Test(dataProvider = "SameUnclipped5primeEnd")
+    public void testStrangeOrderDependentBugVersion(final File input) {
+
+        final AbstractMarkDuplicatesCommandLineProgramTester tester = new ByQualityQuerySortedTester();
+        tester.setRecordsNeedSorting(false);
```

Comment:
How does that work in class inheritance? What would be the signature of setRecordsNeedSorting?